tools/scripts/generate-sponsors.mts

```diff
@@ -2,103 +2,158 @@ import * as fs from 'node:fs';
 import * as path from 'node:path';
 
 import fetch from 'cross-fetch';
+import type { SponsorData } from 'website/src/components/FinancialContributors/types.ts';
 
 import { PACKAGES_WEBSITE } from './paths.mts';
 
-type MemberNodes = {
-  account: {
-    id: string;
-    imageUrl: string;
-    name: string;
-    website: string;
-  };
-  totalDonations: { valueInCents: number };
-}[];
-
 const excludedNames = new Set([
   'Josh Goldberg', // Team member 💖
 ]);
 
 const filteredTerms = ['casino', 'deepnude', 'tiktok'];
 
-const { members } = (
-  (await (
-    await fetch('https://api.opencollective.com/graphql/v2', {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({
-        query: `
-          {
-            collective(slug: "typescript-eslint") {
-              members(limit: 1000, role: BACKER) {
-                nodes {
-                  account {
-                    id
-                    imageUrl
-                    name
-                    website
-                  }
-                  tier {
-                    amount {
-                      valueInCents
-                    }
-                    orders(limit: 100) {
-                      nodes {
-                        amount {
-                          valueInCents
-                        }
+const jsonApiFetch = async <T,>(
+  api: string,
+  options?: RequestInit,
+): Promise<T> => {
+  const url = `https://api.${api}`;
+  const response = await fetch(url, options);
+  const body: unknown = await response.json();
+  if (!response.ok) {
+    console.error({ url, response: { status: response.status, body } });
+    throw new Error('API call failed.');
+  }
+  return body as T;
+};
+
+fs.writeFileSync(
```

Comment:
Please restructure this a bit for readability. 

Maybe,
```ts
const openCollectiveSponsorsPromise = /* long expression */;

const thanksDevSponsorsPromise = /* long expression */;

const mergedSponsors = await Promise.all([openCollectiveSponsorsPromise, thanksDevSponsorsPromise])
     .flat()
     .filter(/* etc */)
     .sort(/* etc */);
     
fs.writeFileSync(
  path.join(PACKAGES_WEBSITE, 'data', 'sponsors.json'),
  `${JSON.stringify(mergedSponsors, null, 2)}\n`,
);
```

I get that it _can_ be written as a 150-line one-liner, since the variables are only used once, but.... please don't do that 🙏